terraform_files/aws/main.tf

```diff
@@ -0,0 +1,54 @@
+terraform {
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = "~> 4.0"
+    }
+  }
+}
+
+provider "aws" {
+  region = "us-east-1"
+}
+
+resource "aws_instance" "masters" {
+  count = var.masters_count
+
+  ami                    = var.ipxe_ami_us_east_1
+  instance_type          = var.master_instance_type
+  subnet_id              = var.aws_subnet
+  vpc_security_group_ids = var.aws_vpc_security_group_ids
+  user_data              = var.ipxe_script
+
+  tags = {
+    Name        = "${var.cluster_name}-master-${count.index}"
+    ClusterName = "${var.cluster_name}"
+    NodeType    = "master"
+  }
+
+  root_block_device {
+    volume_size = 100
+    volume_type = "gp3"
+  }
+}
+
+resource "aws_instance" "workers" {
+  count = var.workers_count
+
+  ami                    = var.ipxe_ami_us_east_1
+  instance_type          = var.worker_instance_type
+  subnet_id              = var.aws_subnet
+  vpc_security_group_ids = var.aws_vpc_security_group_ids
+  user_data              = var.ipxe_script
+
+  tags = {
+    Name        = "${var.cluster_name}-worker-${count.index}"
+    ClusterName = "${var.cluster_name}"
+    NodeType    = "worker"
+  }
+
+  root_block_device {
+    volume_size = 100
```

Comment:
not sure if we can pick a smaller size, but workers can do with 20GB disks iirc

Reply:
smaller, cheaper :)